javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinter.java

```diff
@@ -521,7 +522,10 @@ private static NodeText interpret(Node node, CsmElement csm, NodeText nodeText)
 
             pendingIndentation = false;
             if (element instanceof LexicalDifferenceCalculator.CsmChild) {
-                nodeText.addChild(((LexicalDifferenceCalculator.CsmChild) element).getChild());
+                Node child = ((LexicalDifferenceCalculator.CsmChild) element).getChild();
+                if (child.getComment().isPresent())
+                    nodeText.addToken(getTokenKind(child.getComment().get()), child.getComment().get().toString());
```

Comment:
Optionals are intended to be used with functional style. `ifPresent` would work better here.